### DISCUSSION TITLE
Render mp4 video without GUI (base_cam.json format needed)

### DISCUSSION THREAD

**Discussion by Pavan-kumar-tullimilli** [NVlabs/instant-ngp#886] [Q&A]
I am working on a remote server with V100 gpu. 
Since UI is not available, I am not able to generate "**base_cam.json**" (which is needed to save animations) through testbed.
Is there any way to render an animation (mp4) through script (without GUI)?
Is there any documentation on base_cam.json format?

**Comment by Tom94** [comment]
Hi there, `base_cam.json` is currently undocumented and expected to be generated through the GUI.

If you already know the camera matrices for each frame of your animation, you could generate a new `transforms.json` file with empty filenames and use `run.py --test_transforms <your transforms.json>`.

Alternatively, you can also directly set the camera matrices through the python bindings (see `run.py` for how this is done in the case of the `--test_transforms` argument).

Cheers!
